### PR TITLE
refactor(github): Convert datasources to class form

### PR DESCRIPTION
**Pull request by zharinov**
## Context:

- Ref: #8647

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: lib/datasource/github-tags/index.ts [hunk 1]
```diff
@@ -1,185 +1,147 @@
...
-    logger.debug(
-      { githubRepo: repo, err, registryUrl },
-      'Error getting latest commit from GitHub repo'
-    );
-  }
-  if (!digest) {
-    return null;
+  @cache({
+    ttlMinutes: 10,
+    namespace: 'datasource-github-tags',
```
Comment (JamieMagee) [inline lib/datasource/github-tags/index.ts:50]:
```suggestion
    namespace: `datasource-${GitHubTagsDatasource.id}`,
```

File: lib/datasource/github-tags/index.ts [hunk 2]
```diff
@@ -1,185 +1,147 @@
...
-    const res = (await http.getJson<TagResponse>(url)).body.object;
-    if (res.type === 'commit') {
-      digest = res.sha;
-    } else if (res.type === 'tag') {
-      digest = (await http.getJson<TagResponse>(res.url)).body.object.sha;
-    } else {
-      logger.warn({ res }, 'Unknown git tag refs type');
+  @cache({
+    ttlMinutes: 120,
+    namespace: 'datasource-github-tags',
```
Comment (JamieMagee) [inline lib/datasource/github-tags/index.ts:17]:
```suggestion
    namespace: `datasource-${GitHubTagsDatasource.id}`,
```

File: lib/datasource/github-tags/index.ts [hunk 3]
```diff
@@ -1,185 +1,147 @@
...
+      ? this.getTagCommit(registryUrl, repo, newValue)
+      : this.getCommit(registryUrl, repo);
   }
 
-  const apiBaseUrl = getApiBaseUrl(registryUrl);
-  // tag
-  const url = `${apiBaseUrl}repos/${repo}/tags?per_page=100`;
+  @cache({
+    ttlMinutes: 10,
+    namespace: 'datasource-github-tags',
```
Comment (JamieMagee) [inline lib/datasource/github-tags/index.ts:91]:
```suggestion
    namespace: `datasource-${GitHubTagsDatasource.id}`,
```

**Review by JamieMagee — DISMISSED** [review]

**Comment by viceice** [comment]
needs deconflciting

**Comment by rarkins** [comment]
Needs deconflicting

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**